sqlalchemy-stubs/sql/functions.pyi

```diff
@@ -74,7 +75,22 @@ class GenericFunction(Function[_T], metaclass=_GenericMeta):
     packagenames: Any = ...
     clause_expr: Grouping[Any] = ...
     type: TypeEngine[_T] = ...
-    def __init__(self, *args: Any, **kwargs: Any) -> None: ...
+    @overload
+    def __init__(self, *args: ColumnElement[Any], bind: Optional[Union[Engine, Connection]] = ...,
+                 type_: Type[TypeEngine[_T]]) -> None: ...
```

Comment:
This now causes an error on `functions.concat(users.c.user, "some.suffix")`. But it works at runtime.

Reply:
Since this is the only remaining small issue, I am going to merge this now, you can fix it in a separate PR.